### PR TITLE
Overhaul Contributing to Documentation

### PR DESCRIPTION
**Pull request by stevepiercy**
- Start following Microsoft Style Guide, specifically use Sentence case for headings, not Title Case.
- Rename and refactor docs-writing-guide.md to myst-reference.md, because that is what it really is.
- Break out Sphinx extensions into its own page.
- Reorganize sections into a more logical order.
- Add some Plone-specific documentation tips.

### CODE REVIEW

**Comment by stevepiercy** [comment]
@plone/documentation-team could I get a review please? @ksuess @fredvd @polyester @spereverde

**Comment by stevepiercy** [comment]
The test failure may be ignored. web.archive.org has been suffering with response times recently. We might need to ignore links to `web.archive.org` or remove them from the docs entirely.

File: docs/contributing/authors.md [hunk 1]
```diff
@@ -85,17 +61,18 @@ Spelling is configured in {file}`Makefile`, {file}`.vale.ini`, and in files in `
...
 ```shell
 make vale
 ```
 
-Because it is difficult to automate good English grammar and syntax, we do not strictly enforce it.
+Because it is difficult to automate good American English grammar and syntax, we do not strictly enforce it.
 We also understand that contributors might not be fluent in English.
 We encourage contributors to make a reasonable effort, and to seek help from community members who are fluent in English.
-When you submit a pull request, please select a Reviewer to review your work.
+Please ask!
```
Comment (fredvd) [inline docs/contributing/authors.md:75]: What you could add here that if the author does reasonable effort, 'native speaker' community members can step in to fix grammatical errors.

File: docs/contributing/myst-reference.md [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 ---
 
-(contributing-writing-docs-guide)=
+(contributing-myst-reference)=
 
-# General Guide to Writing Documentation
+# MyST reference
 
-This guide provides general help for writing documentation for Plone.
+This chapter provides information and examples to write proper MyST syntax—with references to Sphinx extensions for their specific directives—in Plone Documentation.
```
Comment (fredvd) [inline docs/contributing/myst-reference.md]: ...and example on how to write .... 
 
as non native speaker I stumble on the omission of 'on how' when I read the sentences but I don't know if others would have this as well

**Review by fredvd — DISMISSED** [review]

**Comment by fredvd** [comment]
Sigh, Github ignored my approval comment.... @stevepiercy it looks excellent, I tried to check the the individual changes while having the netlify preview open as well, but it is difficult for me to check both and see the whole.  I had 2 minor grammar things where I can only comment that as a non native speaker I 'stumble' a bit while reading the sentence.  

not in this PR; I'd switch the very first to sentences of this section: contributions are welcome, this document describes how to do it.  But very minor. 

When this section is merged, please give me a ping and I'll reread it and see if something stands out to me while understanding it.   

Thanks, this is a very solid section now!

File: docs/contributing/authors.md (hunk 1 above)
Comment (stevepiercy) [reply docs/contributing/authors.md:75]: Reworded in 55c82303fe16a0fa794884c9132e099106d28338

File: docs/contributing/myst-reference.md (hunk 2 above)
Comment (stevepiercy) [reply docs/contributing/myst-reference.md]: Fixed in c9b4d4e260a0bdd74f77aaece6c2bdc058f3f198

**Comment by stevepiercy** [comment]
@fredvd added your suggestions, as well as how to use the new Sphinx and MyST extensions for inline images and videos.

**Comment by stevepiercy** [comment]
NOTICE: I'm going to merge this PR on 2023-02-18 (Saturday), with or without further review. It's been sitting here for almost 2 months. If you want to add your review, please do so before then. Thank you!

**Review by fredvd — APPROVED** [review]
lgtm, thank you again.

 I'l pick up the deploy caching part later today.

**Pull request merged**